File: java/src/main/java/com/swiftleap/rules/connector/FieldFormatter.java
package com.swiftleap.rules.connector;

import java.text.SimpleDateFormat;
import java.util.Date;

public interface FieldFormatter {
    String format(Object value);

    class DefaultFieldFormatter implements FieldFormatter {
        SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");

        @Override
        public String format(Object value) {
            if (value == null)
                return null;
            if (value instanceof Date)
                return df.format((Date) value);
            return value.toString().trim();
        }
    }
}

File: dotnet/RulesConnector/FieldFormatter.cs
using System;

namespace SwiftLeap.RulesConnector
{
    public interface FieldFormatter
    {
        string Format(object value);
    }

    public class DefaultFieldFormatter : FieldFormatter
    {
        public string Format(object value)
        {
            switch (value)
            {
                case null:
                    return null;
                case DateTime date:
                    return date.ToString("dd/MM/yyyy");
                default:
                    return value.ToString();
            }
        }
    }
}